### PR TITLE
2 as client i want to get user data

### PR DESCRIPTION
**Pull request by rubeniriso**
Working functionalities + tests for:

- Generic error
- User found
- User not found

### CODE REVIEW

File: tests/app/Infrastructure/Controller/GetUserDataControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        parent::setUp();
+
+        $this->userDataSource = Mockery::mock(UserDataSource::class);
+        $this->app->bind(UserDataSource::class, fn () => $this->userDataSource);
+    }
+
+    /**
+     * @test
+     */
+    public function userWithGivenIdDoesNotExist()
```
Comment (AMBuilds) [inline tests/app/Infrastructure/Controller/GetUserDataControllerTest.php]: Change test name format to snake

**Review by AMBuilds — CHANGES_REQUESTED** [review]
Please, Look at the comments.

File: tests/app/Infrastructure/Controller/GetUserDataControllerTest.php (hunk 1 above)
Comment (rubeniriso) [reply tests/app/Infrastructure/Controller/GetUserDataControllerTest.php]: done, please review.

**Review by AMBuilds — APPROVED** [review]
Ok to merge

**Pull request merged**